includes/class-wc-admin-notes-order-milestones.php

```diff
@@ -0,0 +1,251 @@
+			);
+			$note->set_type( WC_Admin_Note::E_WC_ADMIN_NOTE_INFORMATIONAL );
+			$note->set_icon( 'trophy' );
+			$note->set_name( self::FIRST_ORDER_NOTE_NAME );
+			$note->set_source( 'woocommerce-admin' );
+			$note->add_action( 'learn-more', __( 'Learn more', 'woocommerce-admin' ), 'https://docs.woocommerce.com/document/managing-orders/' );
+			$note->save();
+		}
+
+		if ( 10 === $orders_count ) {
+			// Add the first ten orders note.
+			$note = new WC_Admin_Note();
+			$note->set_title(
+				sprintf(
+					/* translators: Number of orders processed. */
+					__( 'Congratulations on processing %s orders!', 'woocommerce-admin' ),
+					wc_format_decimal( 10 )
+				)
+			);
+			$note->set_content(
+				__( "You've hit the 10 orders milestone! Look at you go. Browse some WooCommerce success stories for inspiration.", 'woocommerce-admin' )
+			);
+			$note->set_type( WC_Admin_Note::E_WC_ADMIN_NOTE_INFORMATIONAL );
+			$note->set_icon( 'trophy' );
+			$note->set_name( self::TEN_ORDERS_NOTE_NAME );
+			$note->set_source( 'woocommerce-admin' );
+			$note->add_action( 'browse', __( 'Browse', 'woocommerce-admin' ), 'https://woocommerce.com/success-stories/' );
+			$note->save();
+		}
+	}
+
+	/**
+	 * Backfill the store's current milestone.
+	 *
+	 * Used to avoid celebrating milestones that were reached before plugin activation.
+	 */
+	public function backfill_last_milestone() {
+		$this->set_last_milestone( $this->get_current_milestone() );
+	}
+
+	/**
+	 * Get the store's last milestone.
+	 *
+	 * @return int Last milestone reached.
+	 */
+	public function get_last_milestone() {
+		return get_option( self::LAST_ORDER_MILESTONE_OPTION_KEY, 0 );
+	}
+
+	/**
+	 * Update the last reached milestone.
+	 *
+	 * @param int $milestone Last milestone reached.
+	 */
+	public function set_last_milestone( $milestone ) {
+		update_option( self::LAST_ORDER_MILESTONE_OPTION_KEY, $milestone );
+	}
+
+	/**
+	 * Calculate the current orders milestone.
+	 *
+	 * Based on the threshold values in $this->milestones.
+	 *
+	 * @return int Current orders milestone.
+	 */
+	public function get_current_milestone() {
+		$milestone_reached = 0;
+		$orders_count      = $this->get_orders_count();
+
+		foreach ( $this->milestones as $milestone ) {
+			if ( $milestone <= $orders_count ) {
+				$milestone_reached = $milestone;
+			}
+		}
+
+		return $milestone_reached;
+	}
+
+	/**
+	 * Add milestone notes for other significant thresholds.
+	 */
+	public function other_milestones() {
+		$last_milestone    = $this->get_last_milestone();
+		$current_milestone = $this->get_current_milestone();
+
+		if ( $current_milestone <= $last_milestone ) {
+			return;
+		}
+
+		$this->set_last_milestone( $current_milestone );
+
+		// We only want one milestone note at any time.
+		WC_Admin_Notes::delete_notes_with_name( self::ORDERS_MILESTONE_NOTE_NAME );
+
+		// Add the milestone note.
+		$note = new WC_Admin_Note();
```

Comment:
All the new note creation in the file here is a bit repetitive - did you consider extracting it into a method on the class here? Or any particular reason not to?

Reply:
The only things repeated are the type, icon, and source so I didn't see the repetition as that big of a problem.